### PR TITLE
Add builders and conversions for cluster operation types

### PR DESCRIPTION
**Pull request by timvisee**
This adds builders and conversions for `MoveShard`, `ReplicateShard`, `AbortShardTransfer` and `Replica`.

image

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you formatted your code locally using `cargo +nightly fmt --all` command prior to submission?
3. [x] Have you checked your code using `cargo clippy --all --all-features` command?

### CODE REVIEW

File: tests/protos.rs [hunk 1]
```diff
@@ -608,6 +608,26 @@ fn configure_builder(builder: Builder) -> Builder {
...
+            ("ReplicateShard.to_peer_id", DEFAULT_OPTION),
+            ("ReplicateShard.method", DEFAULT_OPTION),
+            // AbortShardTransfer
+            ("AbortShardTransfer.shard_id", DEFAULT_OPTION),
+            ("AbortShardTransfer.to_shard_id", DEFAULT_OPTION),
+            ("AbortShardTransfer.from_peer_id", DEFAULT_OPTION),
+            ("AbortShardTransfer.to_peer_id", DEFAULT_OPTION),
+            // Replica
+            ("Replica.shard_id", DEFAULT_OPTION),
+            ("Replica.peer_id", DEFAULT_OPTION),
```
Comment (JojiiOfficial) [inline tests/protos.rs:630]:
```suggestion
            // MoveShard
            ("MoveShard.shard_id", PUBLIC_ONLY),
            ("MoveShard.to_shard_id", DEFAULT_OPTION),
            ("MoveShard.from_peer_id", PUBLIC_ONLY),
            ("MoveShard.to_peer_id", PUBLIC_ONLY),
            ("MoveShard.method", DEFAULT_OPTION_INTO),
            // ReplicateShard
            ("ReplicateShard.shard_id", PUBLIC_ONLY),
            ("ReplicateShard.to_shard_id", DEFAULT_OPTION),
            ("ReplicateShard.from_peer_id", PUBLIC_ONLY),
            ("ReplicateShard.to_peer_id", PUBLIC_ONLY),
            ("ReplicateShard.method", DEFAULT_OPTION_INTO),
            // AbortShardTransfer
            ("AbortShardTransfer.shard_id", PUBLIC_ONLY),
            ("AbortShardTransfer.to_shard_id", DEFAULT_OPTION),
            ("AbortShardTransfer.from_peer_id", PUBLIC_ONLY),
            ("AbortShardTransfer.to_peer_id", PUBLIC_ONLY),
            // Replica
            ("Replica.shard_id", PUBLIC_ONLY),
            ("Replica.peer_id", PUBLIC_ONLY),
```

Comment (timvisee) [reply tests/protos.rs]: Thanks!

**Review by JojiiOfficial — APPROVED** [review]

File: src/builder_ext.rs [hunk 2]
```diff
@@ -314,11 +315,17 @@ impl UpdateCollectionClusterSetupRequestBuilder {
 }
 
 impl MoveShardBuilder {
-    pub fn new(shard_id: u32, from_peer_id: u64, to_peer_id: u64) -> Self {
+    pub fn new(
+        shard_id: u32,
+        from_peer_id: u64,
+        to_peer_id: u64,
+        method: ShardTransferMethod,
```
Comment (timvisee) [inline src/builder_ext.rs:322]: The method should be optional, so we probably want to revert this change.

Instead, we want `UpdateCollectionClusterSetupRequestBuilder.operation` to be required.

Comment (JojiiOfficial) [reply src/builder_ext.rs:322]: Oh my bad, misunderstood your message. I'll revert it and implement it correctly!

Comment (timvisee) [reply src/builder_ext.rs]: No problem. Thanks!
